### PR TITLE
Stabilize ARM template

### PR DESCRIPTION
**Pull request by akhilayesantarao**
- shift create kb to first update call, store kbid in blob
- don't update app settings, use linked template to get http trigger 
- added app insights and option for server farms sku in arm template

### CODE REVIEW

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs [hunk 1]
```diff
@@ -144,13 +147,58 @@ public static class UploadToQnAMaker
...
+                    {
+                        kbId = await streamReader.ReadLineAsync();
+                    }
+                }
+            }
+            else
+            {
+                kbId = await CreateKB(log);
+                MemoryStream stream = new MemoryStream();
+                StreamWriter writer = new StreamWriter(stream);
```
Comment (bingisbestest) [inline CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:174]: using (...) these

File: azuredeploy.json [hunk 2]
```diff
@@ -263,71 +318,28 @@
...
-            "value": "[variables('functionAppName')]"
-        },
-        "Azure Function Host Code": {
-            "type": "string",
-            "value": "[listkeys(concat(resourceId('Microsoft.Web/sites', variables('functionAppName')), '/host/default/'),'2019-08-01').functionKeys.default]"
-        },
         "HTTP Trigger": {
             "type": "string",
-            "value": "[concat('https://',variables('functionAppName'),'.azurewebsites.net/api/init-accelerator?code=',listkeys(concat(resourceId('Microsoft.Web/sites', variables('functionAppName')), '/host/default/'),'2019-08-01').functionKeys.default)]"
+            "value": "[reference('initaccelerator').outputs.URL.value]"
```
Comment (bingisbestest) [inline azuredeploy.json:342]: we can remove this output if not needed.

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs [hunk 3]
```diff
@@ -125,7 +128,7 @@ public static class UploadToQnAMaker
             var stopwatch = new Stopwatch();
```
Comment (bingisbestest) [inline CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:128]: unused ?

File: azuredeploy.json [hunk 4]
```diff
@@ -263,71 +318,28 @@
...
+            "name": "initaccelerator",
             "dependsOn": [
-                "[resourceId('Microsoft.CognitiveServices/accounts', variables('cognitiveServicesAllInOneName'))]",
                 "[resourceId('Microsoft.Web/Sites/sourcecontrols', variables('functionAppName'), 'web')]"
             ],
             "properties": {
                 "mode": "Incremental",
                 "templateLink": {
-                    "uri": "[concat('https://',variables('functionAppName'),'.azurewebsites.net/api/init-kb?code=',listkeys(concat(resourceId('Microsoft.Web/sites', variables('functionAppName')), '/host/default/'),'2019-08-01').functionKeys.default)]",
+                    "uri": "[concat('https://raw.githubusercontent.com/jennifermarsman/cognitive-search-qna-solution/',variables('branch'),'/linkedazuredeploy.json')]",
```
Comment (bingisbestest) [inline azuredeploy.json:328]: Use a variable for this if we want to change the path /jennifer/....

**Comment by bingisbestest** [comment]
Also we need to : 
1. Cache the kbid
2. O/P runtime key for portal

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs (hunk 3 above)
Comment (akhilayesantarao) [reply CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:128]: The time taken by update operation is logged

File: azuredeploy.json (hunk 4 above)
Comment (akhilayesantarao) [reply azuredeploy.json:328]: okay

File: azuredeploy.json (hunk 2 above)
Comment (akhilayesantarao) [reply azuredeploy.json:342]: Need it for the trigger

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs (hunk 1 above)
Comment (akhilayesantarao) [reply CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:174]: okay

**Review by ghost — APPROVED** [review]

**Pull request merged**